go/vt/vtgate/planbuilder/dml.go

```diff
@@ -165,6 +209,21 @@ func generateDeleteSubquery(del *sqlparser.Delete, table *vindexes.Table) string
 	return buf.String()
 }
 
+func generateUpdateSubquery(upd *sqlparser.Update, table *vindexes.Table) string {
+	if len(table.Owned) == 0 {
+		return ""
+	}
+	buf := sqlparser.NewTrackedBuffer(nil)
+	buf.WriteString("select ")
+	prefix := ""
```

Comment:
Some reviewers don't like this pattern. I've been converting to this one as I change stuff:
```
for i, ... {
  if i == 0 {
    buf.Myprintf("%s%v", prefix, cv.Column)
  } else {
    buf.Myprintf(", %s%v", prefix, cv.Column)
  }
}
```